### DISCUSSION TITLE
When fetching a document with a specific property set to an ObjectID, add data from another document

### DISCUSSION THREAD

**Discussion by maziluradu** [TypedProject/ts-express-decorators#2365] [Q&A]
I have a simple mongoose model defined like this:

```@Model()
export class User {
    @Property() public name: string;
    @Property() public dateOfBirth: string;
    @Optional() @ObjectID() @Property() public clonesUserId?: string;
}
```

I would like to implement a behavior where if I try to fetch a document (of this model) from my MongoDB database and that document includes a valid "clonesUserId" property, it combines the data from both documents and returns the combined information.

For example, if I have the following 2 documents:

```
// First user (main)
{
    "_id": "abcdefghijklmnopqrstuvwxyz",
    "name": "John Doe",
    "dateOfBirth": "24-07-1997"
}
```

```
// Second user (secondary)
{
    "_id": "secondary-id",
    "clonesUserId": "abcdefghijklmnopqrstuvwxyz"
}
```

.. when fetching the secondary user, the query actually would return the following:

```
// Data from both users, prioritizing secondary user data (like spread operator)
{
    "_id": "secondary-id",
    "name": "John Doe",
    "dateOfBirth": "24-07-1997",
    "clonesUserId": "abcdefghijklmnopqrstuvwxyz",
}
```

Is there any clean way to do this besides manually adding code every time I am using "userModel.findOne(...)"? I would love if there was an implementation in the model's class only

**Comment by Romakita** [comment]
The question is: mongoose able to do that?

**Reply by Romakita** [reply]
@maziluradu have you found a solution?
